perun-core/src/main/java/cz/metacentrum/perun/core/implApi/modules/attributes/AttributesModuleAbstract.java

```diff
@@ -33,7 +34,7 @@ public AttributeDefinition getAttributeDefinition() {
 		return attr;
 	}
 
-	public List<String> resolveVirtualAttributeValueChange(PerunSessionImpl perunSession, String message) throws InternalErrorException, WrongReferenceAttributeValueException, AttributeNotExistsException, WrongAttributeAssignmentException {
+	public List<AuditerMessage> resolveVirtualAttributeValueChange(PerunSessionImpl perunSession, AuditerMessage message) throws InternalErrorException, WrongReferenceAttributeValueException, AttributeNotExistsException, WrongAttributeAssignmentException {
```

Comment:
I whink it would be better if this method would work with AuditEvents instead of AuditerMessage. The information about session is redundant.